### PR TITLE
[QC-478] Generate the modulefiles for QC

### PR DESCRIPTION
**Pull request by Barthelemy**

### CODE REVIEW

**Comment by Barthelemy** [comment]
@ktf as requested

File: qualitycontrol.sh [hunk 1]
```diff
@@ -106,30 +106,11 @@ fi
...
-            O2/$O2_VERSION-$O2_REVISION                                                             \\
-            ${ARROW_REVISION:+arrow/$ARROW_VERSION-$ARROW_REVISION}                                 \\
-            Control-OCCPlugin/$CONTROL_OCCPLUGIN_VERSION-$CONTROL_OCCPLUGIN_REVISION                \\
-            ${PYTHON_REVISION:+Python/$PYTHON_VERSION-$PYTHON_REVISION}                             \\
-            ${PYTHON_MODULES_REVISION:+Python-modules/$PYTHON_MODULES_VERSION-$PYTHON_MODULES_REVISION}
+alibuild-generate-module --bin --lib > etc/modulefiles/$PKGNAME
 
 # Our environment
+cat >> etc/modulefiles/$PKGNAME <<EOF
+EoF
```
Comment (ktf) [inline qualitycontrol.sh:113]: I think you have one extra EoF.

**Review by sy-c — DISMISSED** [review]

File: qualitycontrol.sh [hunk 2]
```diff
@@ -106,30 +106,11 @@ fi
...
 set QUALITYCONTROL_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
 setenv QUALITYCONTROL_ROOT \$QUALITYCONTROL_ROOT
 prepend-path PATH \$QUALITYCONTROL_ROOT/bin
```
Comment (ktf) [inline qualitycontrol.sh:116]: PATH is already amended correctly if you use the `--bin` option.

File: qualitycontrol.sh [hunk 3]
```diff
@@ -161,3 +142,5 @@ fi
 cat > $INSTALLROOT/.rpm-extra-deps <<EOF
 glfw # because the build machine some times happen to have glfw installed. Then it is necessary to have it in the destination
 EOF
+
+EOF
```
Comment (ktf) [inline qualitycontrol.sh:146]: I think you misplaced the EOF.

File: qualitycontrol.sh [hunk 4]
```diff
@@ -106,30 +106,11 @@ fi
...
-            Configuration/$CONFIGURATION_VERSION-$CONFIGURATION_REVISION                            \\
-            Common-O2/$COMMON_O2_VERSION-$COMMON_O2_REVISION                                        \\
-            ${LIBINFOLOGGER_REVISION:+libInfoLogger/$LIBINFOLOGGER_VERSION-$LIBINFOLOGGER_REVISION} \\
-            FairRoot/$FAIRROOT_VERSION-$FAIRROOT_REVISION                                           \\
-            O2/$O2_VERSION-$O2_REVISION                                                             \\
-            ${ARROW_REVISION:+arrow/$ARROW_VERSION-$ARROW_REVISION}                                 \\
-            Control-OCCPlugin/$CONTROL_OCCPLUGIN_VERSION-$CONTROL_OCCPLUGIN_REVISION                \\
-            ${PYTHON_REVISION:+Python/$PYTHON_VERSION-$PYTHON_REVISION}                             \\
-            ${PYTHON_MODULES_REVISION:+Python-modules/$PYTHON_MODULES_VERSION-$PYTHON_MODULES_REVISION}
+alibuild-generate-module --bin --lib > etc/modulefiles/$PKGNAME
```
Comment (ktf) [inline qualitycontrol.sh:109]: You should probably also add `alice-recipe-tools` as a `build_requires`.

**Review by ktf — CHANGES_REQUESTED** [review]
See inline comments.

**Comment by Barthelemy** [comment]
Thanks, I screwed the EOF during a merge. I fixed the other things as well.

**Comment by TimoWilken** [comment]
@Barthelemy if it helps, here's the error from aliDoctor (scroll to the end):
```
++ timeout -s9 600 aliDoctor --defaults o2-dataflow O2Suite
SUCCESS: Package GCC-Toolchain will be picked up from the system.
WARNING: Package CMake cannot be picked up from the system and will be built by aliBuild.
WARNING: This is due to the fact the following script fails:
WARNING: 
WARNING: brew() { true; }; verge() { [[  "$1" = "`echo -e "$1\n$2" | sort -V | head -n1`" ]]; }
WARNING: type cmake && verge 3.18.0 `cmake --version | sed -e 's/.* //' | cut -d. -f1,2,3`
WARNING: 
WARNING: with the following output:
WARNING: 
WARNING: CMake: /bin/bash: line 1: type: cmake: not found
WARNING: CMake: 
WARNING: 
SUCCESS: Required package openmp will be picked up from the system.
WARNING: Package libuv cannot be picked up from the system and will be built by aliBuild.
WARNING: This is due to the fact the following script fails:
WARNING: 
WARNING: brew() { true; }; printf "#include <uv.h>" | c++ -I$(brew --prefix libuv)/include -xc++ - -c -o /dev/null 2>&1
WARNING: 
WARNING: with the following output:
WARNING: 
WARNING: libuv: <stdin>:1:10: fatal error: uv.h: No such file or directory
WARNING: libuv: compilation terminated.
WARNING: libuv: 
WARNING: 
SUCCESS: Required package MySQL will be picked up from the system.
SUCCESS: Package SWIG will be picked up from the system.
WARNING: Package Python cannot be picked up from the system and will be built by aliBuild.
WARNING: This is due to the fact the following script fails:
WARNING: 
WARNING: brew() { true; }; python3 -c 'import sys; import sqlite3; sys.exit(1 if sys.version_info < (3, 5) else 0)' && pip3 --help > /dev/null && printf '#include "pyconfig.h"' | cc -c $(python-config --includes) -xc -o /dev/null -; if [ $? -ne 0 ]; then printf "Python, the Python development packages, and pip must be installed on your system.\nUsually those packages are called python, python-devel (or python-dev) and python-pip.\n"; exit 1; fi
WARNING: 
WARNING: with the following output:
WARNING: 
WARNING: Python: /bin/bash: pip3: command not found
WARNING: Python: Python, the Python development packages, and pip must be installed on your system.
WARNING: Python: Usually those packages are called python, python-devel (or python-dev) and python-pip.
WARNING: Python: 
WARNING: 
SUCCESS: Required package Python-system will be picked up from the system.
WARNING: Package lz4 cannot be picked up from the system and will be built by aliBuild.
WARNING: This is due to the fact the following script fails:
WARNING: 
WARNING: brew() { true; }; printf "#include <lz4.h>\n" | c++ -xc++ -I$(brew --prefix lz4)/include - -c -M 2>&1
WARNING: 
WARNING: with the following output:
WARNING: 
WARNING: lz4: <stdin>:1:10: fatal error: lz4.h: No such file or directory
WARNING: lz4: compilation terminated.
WARNING: lz4: 
WARNING: 
WARNING: Package ZeroMQ cannot be picked up from the system and will be built by aliBuild.
WARNING: This is due to the fact the following script fails:
WARNING: 
WARNING: brew() { true; }; printf "#include <zmq.h>\n#if(ZMQ_VERSION < 40105)\n#error \"zmq version >= 4.1.5 needed\"\n#endif\n int main(){}" | c++ -I$(brew --prefix zeromq)/include -xc++ - -o /dev/null 2>&1
WARNING: 
WARNING: with the following output:
WARNING: 
WARNING: ZeroMQ: <stdin>:1:10: fatal error: zmq.h: No such file or directory
WARNING: ZeroMQ: compilation terminated.
WARNING: ZeroMQ: 
WARNING: 
WARNING: Package node cannot be picked up from the system and will be built by aliBuild.
WARNING: This is due to the fact the following script fails:
WARNING: 
WARNING: brew() { true; }; node -h &> /dev/null || { echo "node is missing"; exit 1; }
WARNING: if [[ $(printf "%d%04d%04d" $(node --version | sed -e 's/v//g; s/\./ /g')) -le 900090000 ]]; then echo "higher node version required"; exit 1; fi
WARNING: 
WARNING: with the following output:
WARNING: 
WARNING: node: node is missing
WARNING: node: 
WARNING: 
Traceback (most recent call last):
  File "/usr/local/bin/aliBuild", line 132, in <module>
    doMain(args, parser)
  File "/usr/local/bin/aliBuild", line 65, in doMain
    doDoctor(args, parser)
  File "/usr/local/lib/python2.7/dist-packages/alibuild_helpers/doctor.py", line 188, in doDoctor
    log                     = info)
  File "/usr/local/lib/python2.7/dist-packages/alibuild_helpers/utilities.py", line 307, in getPackageList
    dieOnError(spec["package"].lower() != lowerPkg,
TypeError: 'NoneType' object has no attribute '__getitem__'
```

**Comment by Barthelemy** [comment]
thank you, I think that I had a simple typo.

**Comment by Barthelemy** [comment]
something went wrong apparently

**Comment by Barthelemy** [comment]
@TimoWilken I don't understand what is going on. Why are these tests failing ?

**Comment by TimoWilken** [comment]
@Barthelemy:

- the build/AliRoot/alidist one is likely failing because of problems with the CERN GitLab (issue open with CERN IT) -- it doesn't always fail, so I'll mark it as pending so it gets picked up again
- the macOS one has errors at the end of the full log:
  ```
  gyp ERR! stack Error: Could not find any Python installation to use
  ```
  and
  ```
  npm WARN enoent ENOENT: no such file or directory, open '[...]/alidist-o2/sw/BUILD/ac91ffe5ed9f9d9b45b89912907bcdd2b28dc142/qcg/package.json'
  ```
  @MichaelLettrich @ktf any ideas?

**Comment by awegrzyn** [comment]
>   ```
>   gyp ERR! stack Error: Could not find any Python installation to use
>   ```

This error is probably coming from `qcg`, although on my machine with new XCode everything compiles without any issue:

```
gyp info find Python using Python version 2.7.16 found at "/System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python"
```
What's your `python` configuration on macOS builders?

**Comment by Barthelemy** [comment]
@TimoWilken and @awegrzyn  thank you for the help. For the mac I understand. For Ubuntu, I now see
```
++ python3 -m venv /mnt/mesos/sandbox/sandbox/o2-alidist/sw/INSTALLROOT/6857cc6e0029a64e9d3b4e1bdb79846faf34f80f/ubuntu1804_x86-64/Python-modules/1.0-1/share/python-modules
The virtual environment was not created successfully because ensurepip is not
available.  On Debian/Ubuntu systems, you need to install the python3-venv
package using the following command.

    apt-get install python3-venv

You may need to use sudo with that command.  After installing the python3-venv
package, recreate your virtual environment.

Failing command: ['/mnt/mesos/sandbox/sandbox/o2-alidist/sw/INSTALLROOT/6857cc6e0029a64e9d3b4e1bdb79846faf34f80f/ubuntu1804_x86-64/Python-modules/1.0-1/share/python-modules/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']

## End sw/BUILD/Python-modules-latest/log
```
@pzhristov any idea ?

**Comment by TimoWilken** [comment]
@Barthelemy The Ubuntu check is new, I'll install python3-venv in the containers.

**Comment by pzhristov** [comment]
It seems all the changes are done in #2736, closing.

**Pull request closed without merging**